### ISSUE TITLE
Get Invoice API returns throws error rather than returning 404 code.

### ISSUE THREAD

**Issue by thandaanda** [XeroAPI/xero-node#21]
Get Call fails with error:

This is the error response.

`{ Error: GET call failed with: 404
    at self.oa.get.stream (/home/keyur/Documents/Git/xeroAPI/node_modules/xero-node/lib/application.js:327:38)
    at passBackControl (/home/keyur/Documents/Git/xeroAPI/node_modules/xero-node/lib/oauth/oauth.js:406:25)
    at IncomingMessage.<anonymous> (/home/keyur/Documents/Git/xeroAPI/node_modules/xero-node/lib/oauth/oauth.js:422:17)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:188:7)
    at endReadableNT (_stream_readable.js:975:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
  data: { 'The resource you\'re looking for cannot be found': '' } }`

**Comment by dupski** [comment]
Hi @thandaanda - Just had a look at this.

Throwing errors on 404 responses from the API is by design - the API call has not been successful because the resource was not found.

The problem is there was not a clean way to programmatically determine what the actual error was, other than inspecting the Error's `message` property.

I've just raised a PR (https://github.com/XeroAPI/xero-node/pull/153) that addresses this issue.

Once the PR is accepted, you'll be able to do something like:

```js
currentApp.core.contacts
  .getContact(nonExistentId)
  .then(response => {
    // ... do the successful thing
  })
  .catch(err => {
    if (err.statusCode == 404) {
      // ... do something specific for 404 errors...
    }
  });
```

**Comment by philals** [comment]
Hi @thandaanda @dupski 

I have accepted the PR and published to 2.15.0

Thanks for the contribution.
